docs/source/user-guide/transformations/joins.md

```diff
@@ -1,229 +1,268 @@
 
-{{code_block('user-guide/transformations/joins','df6',['DataFrame'])}}
+### Anti join
 
-```python exec="on" result="text" session="user-guide/transformations/joins"
---8<-- "python/user-guide/transformations/joins.py:df6"
-```
+Conversely, an anti join will return the rows of the left dataframe that do not have a match in the right dataframe:
 
-You want to answer this question: which of the cars have had repairs carried out?
+{{code_block('user-guide/transformations/joins','anti-join',['join'])}}
 
-An inner join does not answer this question directly as it produces a `DataFrame` with multiple rows for each car that
-has had multiple repair jobs:
+```python exec="on" result="text" session="transformations/joins"
+--8<-- "python/user-guide/transformations/joins.py:anti-join"
+```
 
-{{code_block('user-guide/transformations/joins','inner2',['join'])}}
+## Non-equi joins
 
-```python exec="on" result="text" session="user-guide/transformations/joins"
---8<-- "python/user-guide/transformations/joins.py:inner2"
-```
+In a non-equi join matches between the left and right dataframes are computed differently.
+Instead of looking for matches on key expressions, we provide a single predicate that determines what rows of the left dataframe can be paired up with what rows of the right dataframe.
 
-However, a semi join produces a single row for each car that has had a repair job carried out.
+For example, consider the following Monopoly players and their current cash:
 
-{{code_block('user-guide/transformations/joins','semi',['join'])}}
+{{code_block('user-guide/transformations/joins','players',[])}}
 
-```python exec="on" result="text" session="user-guide/transformations/joins"
---8<-- "python/user-guide/transformations/joins.py:semi"
+```python exec="on" result="text" session="transformations/joins"
+--8<-- "python/user-guide/transformations/joins.py:players"
 ```
 
-### Anti join
-
-Continuing this example, an alternative question might be: which of the cars have **not** had a repair job carried out?
-An anti join produces a `DataFrame` showing all the cars from `df_cars` where the `id` is not present in
-the `df_repairs` `DataFrame`.
+Using a non-equi join we can easily build a dataframe with all the possible properties that each player could be interested in buying.
+We use the function `join_where` to compute a non-equi join:
 
-{{code_block('user-guide/transformations/joins','anti',['join'])}}
+{{code_block('user-guide/transformations/joins','non-equi',['join_where'])}}
 
-```python exec="on" result="text" session="user-guide/transformations/joins"
---8<-- "python/user-guide/transformations/joins.py:anti"
+```python exec="on" result="text" session="transformations/joins"
+--8<-- "python/user-guide/transformations/joins.py:non-equi"
 ```
 
+You can provide multiple expressions as predicates but they all must use Boolean comparison operators and must refer to columns from both dataframes.
```

Comment:
> all must use Boolean comparison operators 

nit: I would not say the operator is boolean, but that it evaluates to a boolean.

Reply:
Changed; isn't it the same thing, though?

Reply:
On boolean operators I think of operators that take boolean arguments, like `AND`, `XOR` etc. Here we take non-boolean arguments and apply an inequality operation that produces a boolean result.